src/nrnpython/nrnpy_hoc.cpp

```diff
@@ -3102,13 +3189,34 @@ PyObject* nrnpy_hoc() {
     m = PyModule_Create(&hocmodule);
     assert(m);
     Symbol* s = NULL;
-    hocobject_type = (PyTypeObject*) PyType_FromSpec(&nrnpy_HocObjectType_spec);
+    spec = obj_spec_from_name("hoc.HocObject");
+    hocobject_type = (PyTypeObject*) PyType_FromSpec(&spec);
     if (PyType_Ready(hocobject_type) < 0)
         goto fail;
     Py_INCREF(hocobject_type);
-    // printf("AddObject HocObject\n");
     PyModule_AddObject(m, "HocObject", (PyObject*) hocobject_type);
 
+    bases = PyTuple_Pack(1, hocobject_type);
+    Py_INCREF(bases);
+    for (auto name: class_name_list) {
+        // TODO: obj_spec_from_name needs a hoc. prepended
+        auto long_name = std::string("hoc.") + name;
+        spec = obj_spec_from_name(long_name.c_str());
+        pto = (PyTypeObject*) PyType_FromSpecWithBases(&spec, bases);
+        sym_to_type_map[hoc_lookup(name)] = pto;
+        type_to_sym_map[pto] = hoc_lookup(name);
+        if (PyType_Ready(pto) < 0)
+            goto fail;
+        Py_INCREF(pto);
+        PyModule_AddObject(m, name, (PyObject*) pto);
+    }
+    Py_DECREF(bases);
+
+    //printf("done\n");
+    // printf("AddObject HocObject\n");
+    //printf("registering Vector\n");
+    //printf("done\n");
```

Comment:
can remove commented out debugging code, especially since it's no longer accurate